c7n/resources/s3.py

```diff
@@ -2494,3 +2494,95 @@ def process_bucket(self, bucket):
 
         s3.put_bucket_lifecycle_configuration(
             Bucket=bucket['Name'], LifecycleConfiguration={'Rules': config})
+
+
+@actions.register('set-bucket-encryption')
+class SetBucketEncryption(BucketActionBase):
+    """Action enables default encryption on S3 buckets
+
+    `enabled`: boolean Optional: Defaults to True
+    `crypto`: aws:kms | AES256` Optional: Defaults to AES256
+    `key`: arn, or id kms key
+
+    :example:
+
+        .. code-block: yaml
+
+            policies:
+              - name: s3-enable-default-encryption-kms
+                resource: s3
+                actions:
+                  - type: set-bucket-encryption
+                    crypto: aws:kms
+                    key: 1234abcd-12ab-34cd-56ef-1234567890ab
+
+              - name: s3-enable-default-encryption-aes256
+                resource: s3
+                actions:
+                  - type: set-bucket-encryption
+                    enabled: false
+    """
+
+    schema = {
+        'type': 'object',
+        'additionalProperties': False,
+        'properties': {
+            'type': {'enum': ['set-bucket-encryption']},
+            'enabled': {'type': 'boolean'},
```

Comment:
Can you follow the same pattern used in set-inventory.  Instead of bool, use enabled/disabled enum to keep consistency.

Reply:
`set-inventory` seems to have three states, `'enabled', 'disabled', 'absent'`, `get-bucket-encryption` will return either a 200 (encryption is enabled) or 404 (encryption is missing). There are other instances of `'enabled': {'type': 'boolean'}` in the `ToggleVersioning` and `ToggleLogging` actions

Is there a reason to handle this via string comparison vs boolean eval that I'm not seeing?

Reply:
I think the main point is a sense of consistency between the actions.  Having to remember to delete bucket encryption is to set enabled to a boolean value and delete inventory is to set state to absent is a bit unwieldily.